### PR TITLE
issue-42

### PR DESCRIPTION
**Pull request by jonathanbratt**
### Documentation:
- [x] Is your feature included in a vignette?
- [x] Does the Roxygen block have both a title and a description?
- [x] Are all parameters documented, including expected classes of those parameters?
- [x] Is the return value documented?
- [x] Is there at least one example?
- [x] Do the vignette(s) look how you'd expect in RStudio (you may have to run `devtools::install(build_vignettes = TRUE)` to see them).
- [x] Does the help documentation look how you'd expect in RStudio?

### Code Quality:
- **not using MMR**
- [x] Can you easily determine what each section of code does (with comments explaining intent)?
- [x] Is your function a verb?
- [x] Is the parameter that's most likely to be piped into this function the first parameter?

### Tests:
- [x] Do you have tests for everything described in the issue you're addressing?
- [x] Did all tests pass (uncommenting any tests you might have commented out while working on things)?
- **all new code covered by tests**

### Housekeeping:
- **all new code satisfies gp**
- [x] Did you increment the version number?
- [x] Did you re-build and document the package at the very end to make sure all changes were picked up?

### Housekeeping:
- [x] Did you review the author checklist (especially things in bold)?

### Documentation:
- [x] Does the vignette explain the context in which this feature would be used?
- [x] Can you understand what this feature is supposed to do from its help?
- [x] Does the example seem like a typical use case?

### Code Quality:
- [x] Can you easily determine what each section of code does (with comments explaining intent)?

### Tests:
- [x] Do the tests match the JIRA issue?
- [x] Can you think of any missing tests?

### CODE REVIEW

**Comment by jonathanbratt** [comment]
going to do it right this time: closes #42 
:)

File: R/extract_features.R [hunk 1]
```diff
@@ -421,6 +421,9 @@ input_fn_builder_EF <- function(features,
 #' potentially be used as features in downstream tasks.)
 #'
 #' @param examples List of \code{InputExample_EF}s to convert.
+#' @param ckpt_dir Character; path to checkpoint directory. If specified,
```
Comment (jonthegeek) [inline R/extract_features.R]: Remove ckpt_dir documentation from this file. Find/replace cp_dir in utils.R to ckpt_dir. Add `@inheritParams find_vocab` to this one.

Likewise, inheritParams FullTokenizer (which should inheritParams load_vocab).

bert_config_from_json_file should rename its arg to bert_config_file, and then extract_features should inheritParams from bert_config_from_json_file.

init_checkpoint should be documented in one place, too, and then the other things should inherit. I'm not sure if `.model_fn_builder_EF` should be the "home", or one of the other functions.

Basically... I really really think we should focus on documenting each repeated-use argument in ONE place (unless it has subtly different meanings in different places), to avoid headaches when we find a typo in one of them, etc.

File: tests/testthat/test-tokenization.R [hunk 2]
```diff
@@ -127,3 +127,17 @@ test_that("is_punctuation correctly classifies characters", {
   testthat::expect_false(is_punctuation("A"))
   testthat::expect_false(is_punctuation(" "))
 })
+
+
+test_that("tokenize_text works correctly", {
+  text <- c("Who doesn't like tacos?", "Not me!")
```
Comment (jonthegeek) [inline tests/testthat/test-tokenization.R]: Does it really work "correctly" if it tokenizes "tacos" the way it does??? (:

File: R/tokenization.R [hunk 3]
```diff
@@ -656,3 +656,73 @@ apply_to_chars <- function(text, .f, ...) {
...
+
+# tokenize_text -----------------------------------------------------------
+
+#' Tokenize Text with Word Pieces
+#'
+#' Given some text and a word piece vocabulary, tokenizes the text. This is
+#' primarily a tool for quickly checking the tokenization of a piece of text.
+#'
+#' @param text Character vector; text to tokenize.
+#' @param ckpt_dir Character; path to checkpoint directory. If specified,
```
Comment (jonthegeek) [inline R/tokenization.R]: Same inheritParams suggestions as above.

File: R/extract_features.R (hunk 1 above)
Comment (jonathanbratt) [reply R/extract_features.R]: I completely agree with this idea. In fact, I had originally used `inheritParams` in `tokenize_text` instead of repeating the documentation. However,  I then realized that the documentation was slightly different (in mentioning just the vocab vs. all three bits of the checkpoint). So I need to think a bit about how to document so that it's "portable" without losing helpful detail (I've almost got it).

It also sounds like there's a lot more documentation clean-up we should do, which should probably be part of its own PR. I'll clean up the args for the functions added in this PR, though.

File: R/tokenization.R (hunk 3 above)
Comment (jonathanbratt) [reply R/tokenization.R]: Yup, this is the place I *did* have it, then took it out.

**Comment by jonathanbratt** [comment]
I've made some of the changes discussed, but (as I said above) I'm holding off on a broader restructuring of the documentation for another PR.
